lib/Sema/CSSimplify.cpp

```diff
@@ -9221,13 +9221,6 @@ bool ConstraintSystem::simplifyAppliedOverloadsImpl(
       for (auto *choice : choices.slice(1))
         choice->setDisabled();
     }
-
-    // Don't attempt further optimization in "diagnostic mode" because
-    // in such mode we'd like to attempt all of the available overloads
-    // regardless of problems related to missing or extraneous labels
-    // and/or arguments.
```

Comment:
I was thinking about this a little more and wondering - this is going to make the solver totally skip over overloads with labeling failures if it's able to fix up a different overload with correct labels, right? What's going to happen now if there's 1 label mismatch and N type mismatches in the expression?

Reply:
If there are overloads that have correct labels in expected positions than they'd be checked and ranked based on the number/kind of type failures they have, the rest is skipped, if there are no matching overloads - all of the overloads are going to be checked to determine what is going on (maybe there are out-of-order arguments or extraneous labels). The reasoning behind this is based on to the fact that name with incorrect labels shouldn't be found by the lookup anyway since labels are part of the name.

Reply:
> The reasoning behind this is based on to the fact that name with incorrect labels shouldn't be found by the lookup anyway since labels are part of the name.

That's true for a compound name, but not a function call. I believe lookup for a function call just uses the bare name (e.g. to get function calls working with `$` labels for SE-0293, all I had to do was change argument-to-parameter label matching)

Reply:
It's the case for all the names, since labels attached to the name in ABI.

Reply:
That said lookup works on the bare name still and most likely will do so in the future.

Reply:
I understand that labels are part of the name for _all_ names. I'm saying that lookup sometimes does not include the name, and instead relies on argument-to-parameter matching.

Reply:
What I am asking is - what diagnostic will be produced for code like this, which has only 1 label mismatch but several type mismatches?

```swift
func test(arg1: Int, arg2: Int) {}
func test(arg: String, arg2: String) {}

test(arg1: "", arg2: "")
```

Reply:
I understand what the concern is here, that's why it is working the way it is :) The idea behind these changes is that matching labels is strong enough indication that developer wanted a particular overload even though the types might not line up at the moment (e.g. they have been filled by completion with holes), so in your example it's going to be two type failures related to arguments instead of one label failure.

That said - there is still a possibility (and that's what your example indicates as well) that one of the labels has a typo, I think we could detect that during label matching and allow such overloads to be attempted in the diagnostic mode together with the ones where labels line up exactly.